### PR TITLE
parser support for create drop ingestion rules

### PR DESCRIPTION
**Pull request by m-kharbat**

### CODE REVIEW

File: sql-parser/src/main/antlr/SqlBase.g4 [hunk 1]
```diff
@@ -71,6 +71,7 @@ statement
     | DROP FUNCTION (IF EXISTS)? name=qname
         '(' (functionArgument (',' functionArgument)*)? ')'                          #dropFunction
     | DROP USER (IF EXISTS)? name=ident                                              #dropUser
+    | DROP INGEST RULE rule_name=qname                                               #dropIngestRule
```
Comment (seut) [inline sql-parser/src/main/antlr/SqlBase.g4]: I think we should use `ident` here.

File: sql-parser/src/main/antlr/SqlBase.g4 [hunk 2]
```diff
@@ -420,6 +421,10 @@ createStmt
         LANGUAGE language=parameterOrIdent
         AS body=parameterOrString                                                    #createFunction
     | CREATE USER name=ident                                                         #createUser
+    | CREATE INGEST RULE rule_name=qname
+        ON source_ident=qname
+        (where)?
+        INTO table_ident=qname                                                       #createIngestRule
```
Comment (seut) [inline sql-parser/src/main/antlr/SqlBase.g4]: also here, I think `rule_name` and `source_ident` should be `ident`

File: sql-parser/src/main/java/io/crate/sql/tree/CreateIngestRule.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return MoreObjects.toStringHelper(this)
+            .add("rule", rule)
+            .add("source", source)
+            .add("target", target)
+            .add("where", where)
+            .toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/tree/CreateIngestRule.java]: is there a reason why `hashCode` and `equals` implementation is needed here?

File: sql-parser/src/main/java/io/crate/sql/tree/DropIngestRule.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        return name;
+    }
+
+    @Override
+    public int hashCode() {
+        return name.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/tree/DropIngestRule.java]: also here, do we need that?

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 5]
```diff
@@ -415,6 +415,21 @@ public void testGrantOnTablePrivilegeStmtBuilder() {
         printStatement("grant ALL ON TABLE my_schema.t, banana.b, tree to anna, nyan, cat");
     }
 
+
+    @Test
+    public void testCreateIngestRuleStmtBuilder() {
+        printStatement("CREATE INGEST RULE v4 ON mqtt WHERE topic like 'v4/%' INTO my_schema.raw_v4");
+        printStatement("CREATE INGEST RULE \"http_r1\" ON \"http\" WHERE topic like 'v4/%' INTO my_schema.raw_v4");
```
Comment (seut) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: afaik `http_r1` can be used without quoting, we should use some example here where the quoting is important like e.g. uppercase idents.

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 6]
```diff
@@ -415,6 +415,21 @@ public void testGrantOnTablePrivilegeStmtBuilder() {
...
+        printStatement("CREATE INGEST RULE v5 ON http INTO t1");
+    }
+
+    @Test
+    public void testDropIngestRuleStmtBuilder() {
+        printStatement("DROP INGEST RULE v4");
+        printStatement("DROP INGEST RULE \"http_r1\"");
```
Comment (seut) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: same here

File: sql-parser/src/main/java/io/crate/sql/tree/DropIngestRule.java (hunk 4 above)
Comment (m-kharbat) [inline sql-parser/src/main/java/io/crate/sql/tree/DropIngestRule.java]: yes, we're extending the abstract statement that extends the abstract class node, so these methods should be implemented

Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/tree/DropIngestRule.java]: afaik these methods are never used (e.g. instances are not used in any list or map). but ok, all implementations currently do, so maybe I missed something...

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 7]
```diff
@@ -419,14 +419,16 @@ public void testGrantOnTablePrivilegeStmtBuilder() {
     @Test
     public void testCreateIngestRuleStmtBuilder() {
         printStatement("CREATE INGEST RULE v4 ON mqtt WHERE topic like 'v4/%' INTO my_schema.raw_v4");
-        printStatement("CREATE INGEST RULE \"http_r1\" ON \"http\" WHERE topic like 'v4/%' INTO my_schema.raw_v4");
+        printStatement("CREATE INGEST RULE \"http_R1\" ON \"http\" WHERE topic like 'v4/%' INTO my_schema.raw_v4");
+        printStatement("CREATE INGEST RULE \"nyan cat space übercool\" ON \"http\" WHERE topic like 'v4/%' INTO my_schema.raw_v4");
```
Comment (seut) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: don't think we need 2 tests for validating that this is an `ident`...

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 8]
```diff
@@ -419,14 +419,16 @@ public void testGrantOnTablePrivilegeStmtBuilder() {
...
         printStatement("CREATE INGEST RULE v5 ON http INTO t1");
     }
 
     @Test
     public void testDropIngestRuleStmtBuilder() {
         printStatement("DROP INGEST RULE v4");
-        printStatement("DROP INGEST RULE \"http_r1\"");
+        printStatement("DROP INGEST RULE \"http_R1\"");
+        printStatement("DROP INGEST RULE \"nyan cat\"");
```
Comment (seut) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: same here, one is enough

**Pull request merged**